src/compiler/checker.ts

```diff
@@ -2010,7 +2010,7 @@ namespace ts {
             }
 
             function getAccessibleSymbolChainFromSymbolTableWorker(symbols: SymbolTable, visitedSymbolTables: SymbolTable[]): Symbol[] {
-                if (contains(visitedSymbolTables, symbols)) {
+                if (contains<SymbolTable>(visitedSymbolTables, symbols)) {

```

Comment:
There are problems with inferring type arguments -- when you get `Foo[]` and expect `ReadonlyArray<T>`, we have trouble inferring that `T = Foo`.